### PR TITLE
build(cargo): bump clap `2` -> `4.5`

### PR DESCRIPTION
**Pull request by cafkafk**
This PR bumps clap from version `2` to `4.5` in order to mitigate
vulnerabilities introduced by clap `>2` on `atty`. This requires some
refactoring. 

Part of this also includes setting a MSRV to rust `1.75.0`, ~and adding a flake
input for `nixpkgs-unstable`~ setting the nixpkgs version to unstable, which is 
the earliest current editon including rustc `1.75.0`.

I'll also attempt to slightly refactor the current structure of clap parsing
out into its own file, perhaps a module, I'll see.

I expect to use the builder pattern, as it's slightly more powerful (at least,
I am pretty sure that's the case), and it's the one I personally prefer. It's
not as bad as in v2 and v3 with unwraps IIRC.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -68,11 +68,11 @@
     devShell.${system} = with pkgs; mkShell {
       inputsFrom = [ self.defaultPackage.${system} ];
       nativeBuildInputs = [
-        cargo
-        rustc
         nix
         rustfmt
         just
+        cargo
+        rustc
```
Comment (cafkafk) [inline flake.nix:75]: I'll revert this, it's just noise

**Comment by srhb** [comment]
I'd like to simply test this on a wharfix node, however, I'm unsure how we can override the rust toolchain to a MSRV-compatible rust toolchain with crane (I'm not sure there even is a simple method) Is there a clap version that solves most of these issues but support a NixOS 23.11 rust toolchain? If that's too complex, I guess we'll have to figure out alternative overlays in our deployments overlays

**Comment by cafkafk** [comment]
> I'd like to simply test this on a wharfix node, however, I'm unsure how we can override the rust toolchain to a MSRV-compatible rust toolchain with crane (I'm not sure there even is a simple method) Is there a clap version that solves most of these issues but support a NixOS 23.11 rust toolchain? If that's too complex, I guess we'll have to figure out alternative overlays in our deployments overlays

IIRC there is some clap 4.x that supports rustc 1.73 (which is probably the easiest solution)

**Comment by cafkafk** [comment]
> IIRC there is some clap 4.x that supports rustc 1.73 (which is probably the easiest solution)

I was able to do this in https://github.com/DBCDK/faythe/pull/53, I'll do the same for this PR.

---

I noticed a problem with `contains_id` being used on boolean flags, where `get_one` should have been used in this PR, which MUST be fixed before merge.

**Comment by cafkafk** [comment]
Superseded by #107

**Pull request closed without merging**